bambou/nurest_object.py

```diff
@@ -352,7 +354,7 @@ def validate(self):
                 continue  # without error
 
             if type(value) != attribute.attribute_type:
-                if attribute.attribute_type != str or type(value) != unicode:
+                if attribute.attribute_type != str or type(value) != str:
```

Comment:
I'm not sure that does what we expect in python 2. `value` is provided by users, will most likely be a python2 `str`. So `type(value)` will always be different than `builtins.str`. Unless you convert users input in the attribute setter?
  
```
In [3]: def is_text(value):          
   ...:     from builtins import str 
   ...:     return type(value) == str
   ...:                              
   ...:                              
                                     
In [4]: is_text(str('coucou'))       
Out[4]: False                        
```

Reply:
Looking at this, i'm pretty sure the validation is good:
https://gist.github.com/pdellaert/026ab1da148f301935af436a3eed145c
(and yes, enterprise.name is a str type attribute

Reply:
@little-dude is right. The validation can fail with the current code. I have an idea on how to fix it and still be python2/3 compatible. Will provide a commit tomorrow

Reply:
Cool, thanks!